src/weathergen/model/model.py

```
@@ -321,110 +321,114 @@ def create(self) -> "Model":
+                            coord_dim = self.geoinfo_sizes[i_stream] * si["token_size"]
+                            dims_embed = torch.tensor(
+                                [dim_out // 2**i for i in range(num_layers - 1, -1, -1)] + [dim_out]
+                            )
+                            dims_embed[dims_embed < coord_dim] = dims_embed[
+                                torch.where(dims_embed >= coord_dim)[0][0]
+                            ]
+                            dims_embed = dims_embed.tolist()
+                        else:
+                            dims_embed = torch.linspace(
+                                dim_embed, dim_out, num_layers + 1, dtype=torch.int32
+                            ).tolist()
+
+                    if is_root():
+                        logger.info("{} :: coord embed: :: {}".format(si["name"], dims_embed))
+
+                    dim_coord_in = self.targets_coords_size[i_stream]
+
+                    # embedding network for coordinates
+                    if etc["net"] == "linear":
+                        self.embed_target_coords[stream_name] = NamedLinear(
+                            f"embed_target_coords_{stream_name}",
+                            in_features=dim_coord_in,
+                            out_features=dims_embed[0],
+                            bias=False,
+                        )
+                    elif etc["net"] == "mlp":
+                        self.embed_target_coords[stream_name] = MLP(
+                            dim_coord_in,
+                            dims_embed[0],
+                            hidden_factor=8,
+                            with_residual=False,
+                            dropout_rate=dropout_rate,
+                            norm_eps=self.cf.mlp_norm_eps,
+                            stream_name=f"embed_target_coords_{stream_name}",
+                        )
+                    else:
+                        assert False
```

nit:  this is not informative. Can you either raise an exception with a full message or, if you prefer asserts, do :

```py
assert etc["net"] in ["mlp", "linear"], etc["net"]
if etc["net"] == "linear":
  ...
```

then the value that is causing problem is clear

---

It's unrelated to the PR. We can do this as part of #1537